x-pack/plugins/security_solution/common/detection_engine/schemas/common/schemas.ts

```diff
@@ -521,3 +521,68 @@ export type Note = t.TypeOf<typeof note>;
 
 export const noteOrUndefined = t.union([note, t.undefined]);
 export type NoteOrUndefined = t.TypeOf<typeof noteOrUndefined>;
+
+export const indexRecord = t.record(
+  t.string,
+  t.type({
+    all: t.boolean,
+    maintenance: t.boolean,
+    manage_ilm: t.boolean,
+    read: t.boolean,
+    create_index: t.boolean,
+    read_cross_cluster: t.boolean,
+    index: t.boolean,
+    monitor: t.boolean,
+    delete: t.boolean,
+    manage: t.boolean,
+    delete_index: t.boolean,
+    create_doc: t.boolean,
+    view_index_metadata: t.boolean,
+    create: t.boolean,
+    manage_follow_index: t.boolean,
+    manage_leader_index: t.boolean,
+    write: t.boolean,
+  })
+);
+export type IndexRecord = t.TypeOf<typeof indexRecord>;
+
+export const indexType = t.type({
+  index: indexRecord,
+});
+export type IndexType = t.TypeOf<typeof indexType>;
+
+export const privilege = t.type({
```

Comment:
These are in a mock file but wonder if we could consolidate with `x-pack/plugins/lists/server/routes/read_privileges_route.mock.ts`

Reply:
I'm wondering if we should since that file explicitly looks for the lists index and has some properties set to some very specific values such as `manage_own_api_key: false`.